torch/_inductor/triton_helpers.py

```diff
@@ -269,15 +272,22 @@ def exclusive_scan_decoupled_lookback(
             flag = unpack_flag(pack, DTYPE_VALUE_AS_UINT)
 
         value = unpack_value(pack, DTYPE_VALUE, DTYPE_VALUE_AS_UINT)
-        exclusive_prefix = combine_fn(value, exclusive_prefix)
+        if prefix_valid:
+            exclusive_prefix = combine_fn(value, exclusive_prefix)
+        else:
+            exclusive_prefix = value
+            prefix_valid = True
```

Comment:
Did you check whether this generates alright code in triton? If not, we can always unroll the first iteration.

Reply:
I checked for cumsum and the llvm IR looks like this
```llvm

  %112 = phi float [ -0.000000e+00, %.lr.ph ], [ %121, %select.unfold ]
  ; ... snip ...

select.unfold:                                    ; preds = %115
  %extract.le = lshr i64 %117, 32
  %extract.t.le = trunc i64 %extract.le to i32
  %120 = bitcast i32 %extract.t.le to float, !dbg !44
  %121 = fadd float %112, %120, !dbg !45
  %122 = icmp ne i32 %118, 2, !dbg !46
  %123 = add nsw i32 %111, -1, !dbg !47
  %124 = icmp sgt i32 %111, 0, !dbg !39
  %125 = select i1 %122, i1 %124, i1 false, !dbg !39
  br i1 %125, label %110, label %select.unfold._crit_edge, !dbg !39
```

Where we can see `%112` is initialized to `0.0e+0` and unconditionally updated with `fadd float %112, %120` inside the loop body. We can also see that updating `test_target` below has been transformed from an if-else to a `select` operation.

For `cumprod` I see an almost identical kernel, except that `%121` is initialized to `1.0e+0` and `fadd` becomes `fmul`. So it seems that llvm was able to infer the correct identity and setup the loop appropriately.

Reply:
That's quite cool actually. I wonder whether it's able to figure out what to do in more complex cases, but this is certainly good enough for now.